src/odemis/driver/ueye.py

```diff
@@ -678,8 +717,25 @@ def __init__(self, name, role, device=None, **kwargs):
 
             self._metadata[model.MD_DET_TYPE] = model.MD_DT_INTEGRATING
 
-            res = (sensorinfo.nMaxWidth, sensorinfo.nMaxHeight)
-            self._metadata[model.MD_SENSOR_SIZE] = self._transposeSizeToUser(res)
+            # compare the max_res parameter and the resolution of the camera sensor
+            # if the max_res param is smaller image cropping will be set through AOI
+            self._sensor_res = (sensorinfo.nMaxWidth, sensorinfo.nMaxHeight)
+            sensor_res_user = self._transposeSizeToUser(self._sensor_res)
+
+            if max_res is None:
+                max_res = self._sensor_res
+            else:
+                max_res = tuple(max_res)
+            if not all(1 <= mr <= r for mr, r in zip(max_res, sensor_res_user)):
+                raise ValueError(f"max_res has to be between (1, 1) and {sensor_res_user}, but got {max_res}")
+            max_res_hw = self._transposeSizeFromUser(max_res)
+
+            # setting a specific Area Of Interest will force image cropping
+            if sensor_res_user != max_res:
+                self.rect_aoi = IS_RECT()  # create the RECT structure for setting the AOI
+                self._set_aoi(sensor_res_user, max_res)
```

Comment:
You shouldn't pass "user" resolution, but the hardware resolution. That's already converted/transposed in the proper format.
I'd suggest to directly compute rect_aoi here, and call SetAOI(). Something like that:
```python
            rect_aoi = IS_RECT() 
            rect_aoi.s32X = (self._sensor_res[0] - max_res_hw[0]) // 2
            rect_aoi.s32Y = (self._sensor_res[1] - max_res_hw[1]) // 2
            rect_aoi.s32Width = max_res_hw[0]
            rect_aoi.s32Height = max_res_hw[1]
            self.SetAOI(rect_aoi)
```

You're done, you can drop the complex `_set_aoi()`!